packages/strapi-utils/lib/sanitize-entity.js

```diff
@@ -21,12 +21,21 @@ const sanitizeEntity = (dataSource, options) => {
 
   const { attributes } = model;
   const allowedFields = getAllowedFields({ includeFields, model, isOutput });
+  const privateAttributes = _.union(
+    _.get(model, 'options.privateAttributes', []),
+    _.get(model, 'options.ignoreGlobalPrivateAttributes', false)
+      ? []
+      : strapi.config.get('api.responses.privateAttributes', [])
+  );
 
   const reducerFn = (acc, value, key) => {
     const attribute = attributes[key];
     const allowedFieldsHasKey = allowedFields.includes(key);
 
-    if (shouldRemoveAttribute(attribute, { withPrivate, isOutput })) {
+    if (
+      shouldRemoveAttribute(attribute, { withPrivate, isOutput }) ||
+      (privateAttributes.includes(key) && isOutput && !withPrivate)
```

Comment:
You could extract this logic to the `shouldRemoveAttribute` function by passing `privateAttributes` in the options and refactoring this condition: `const shouldRemovePrivate = !withPrivate && isOutput;`